### PR TITLE
Post about crash reporter

### PR DESCRIPTION
**Pull request by tikurahul**
Here is a first draft on the post about the crash reporter.

### CODE REVIEW

**Comment by Frijol** [comment]
Yay! I'll take a look today :)

File: 2016-08-21-Crash-Reporter.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+### The Tessel Crash Reporter service
+
+The Tessel 2 [crash reporter](http://crash-reporter.tessel.io) is a web service which automatically collects anonymized crash reports when a crash is detected in the Tessel 2's command line interface (CLI). Given that the CLI is one of the primary ways of interacting with your Tessel it is important that the CLI be extremely stable and a joy to use.
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: "interacting with your Tessel" insert comma "it is important..."

File: 2016-08-21-Crash-Reporter.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+This project started during the early days of the CLI. We wanted to make sure that the team was proactively looking at, and fixing issues that users could potentially run into. We wanted the onboarding experience was as smooth as possible for new users.
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: remove comma after "proactively looking at"

Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: "could potentially" is redundant, and I think overly tentative. How about "...and fixing issues as users encountered them"

Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: Typo in the last sentence, a word is missing

File: 2016-08-21-Crash-Reporter.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+#### How does it work
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: "How it works" perhaps to be declarative like the other headings

File: 2016-08-21-Crash-Reporter.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+Every time you use the Tessel 2 CLI, a central controller dispatches your command to an appropriate task handler. The crash reporter automatically checks for uncaught `Errors` and unresolved `Promises` in the node process that's executing the task.
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: Touch on the privacy issues we worked around (users can decide to turn it on/off, their choice)?

File: 2016-08-21-Crash-Reporter.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+The Tessel 2 [crash reporter](http://crash-reporter.tessel.io) is a web service which automatically collects anonymized crash reports when a crash is detected in the Tessel 2's command line interface (CLI). Given that the CLI is one of the primary ways of interacting with your Tessel it is important that the CLI be extremely stable and a joy to use.
+
+This project started during the early days of the CLI. We wanted to make sure that the team was proactively looking at, and fixing issues that users could potentially run into. We wanted the onboarding experience was as smooth as possible for new users.
+
+#### How does it work
+
+Every time you use the Tessel 2 CLI, a central controller dispatches your command to an appropriate task handler. The crash reporter automatically checks for uncaught `Errors` and unresolved `Promises` in the node process that's executing the task. 
+
+Once the crash reporter detects a crash, it removes all PII (personally identifiable information) from the stack trace, and uploads it to the Crash reporter web service with the user's permission. Below is an example of what that interaction looks like:
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: Do you use the acronym PII later in the post? If not, why identify the acronym?

Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: "and uploads it to the Crash reporter web service with the user's permission" sounds like we will prompt the user each time

File: 2016-08-21-Crash-Reporter.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+```javascript
+ERR! Detected CLI crash [Error: Testing the crash reporter] Error: Testing the crash reporter
+    at CrashReporter.off.CrashReporter.on.CrashReporter.sanitize.CrashReporter.sanitize.redactions.CrashReporter.prompt.CrashReporter.submit.CrashReporter.prompt.then.CrashReporter.post.CrashReporter.status.CrashReporter.test (/Users/rahulrav/Workspaces/Tessel2/CLI/lib/crash-reporter.js:177:25)
+    at process._tickCallback (node.js:406:9)
+    at Function.Module.runMain (module.js:449:11)
+    at startup (node.js:141:18)
+    at node.js:933:3
+INFO Detected a crash in the CLI. Submit crash (y/n) ?
+   If yes(y), subsequent crashes will be submitted automatically.
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md:27]: Is this our current message? Would be a good idea to include a note here on how to turn off the default later

File: 2016-08-21-Crash-Reporter.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+#### The service
+
+The Crash reporter web service collects the crash report, and de-duplicates it using the Sim Hash algorithm. Sim Hash looks at the contents of the crash report and computes a unique signature (called the `fingerprint`) and also keeps track of the `frequency` of the occurence of the report. The service also provides a way to look at `Trending` crashes and also provides a full text `Search` API. The service itself is built using [Google App Engine](https://cloud.google.com/appengine/docs) and the Python Runtime. The full source code for the service is [here](https://github.com/tessel/t2-crash-reporter). 
+
+#### Hi from tessel-crashbot
+
+[Recently](https://github.com/tessel/t2-crash-reporter/issues/15) we landed a really cool feature in the Crash reporter service. In an effort to simplify the worflow of the Tessel team, which manages all feature requests and bug-fixes using GitHub [issues](https://github.com/tessel/t2-cli/issues); the service uses GitHub's API to automatically create and manage issues for every crash report. Anytime there is a new crash in the wild, the ever friendly
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: comma after Recently

File: 2016-08-21-Crash-Reporter.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+#### The service
+
+The Crash reporter web service collects the crash report, and de-duplicates it using the Sim Hash algorithm. Sim Hash looks at the contents of the crash report and computes a unique signature (called the `fingerprint`) and also keeps track of the `frequency` of the occurence of the report. The service also provides a way to look at `Trending` crashes and also provides a full text `Search` API. The service itself is built using [Google App Engine](https://cloud.google.com/appengine/docs) and the Python Runtime. The full source code for the service is [here](https://github.com/tessel/t2-crash-reporter). 
+
+#### Hi from tessel-crashbot
+
+[Recently](https://github.com/tessel/t2-crash-reporter/issues/15) we landed a really cool feature in the Crash reporter service. In an effort to simplify the worflow of the Tessel team, which manages all feature requests and bug-fixes using GitHub [issues](https://github.com/tessel/t2-cli/issues); the service uses GitHub's API to automatically create and manage issues for every crash report. Anytime there is a new crash in the wild, the ever friendly 
+`tessel-crashbot` is ready to help proactively keep track of the problem.
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: note the "every x times it happens it re-posts a comment" feature? I think this is really cool

File: 2016-08-21-Crash-Reporter.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+#### Hi from tessel-crashbot
+
+[Recently](https://github.com/tessel/t2-crash-reporter/issues/15) we landed a really cool feature in the Crash reporter service. In an effort to simplify the worflow of the Tessel team, which manages all feature requests and bug-fixes using GitHub [issues](https://github.com/tessel/t2-cli/issues); the service uses GitHub's API to automatically create and manage issues for every crash report. Anytime there is a new crash in the wild, the ever friendly 
+`tessel-crashbot` is ready to help proactively keep track of the problem. 
+
+![Image of crashbot submission](images/crash_bot.png).
```
Comment (Frijol) [inline 2016-08-21-Crash-Reporter.md]: @reconbot @tcr opinions on how we host images?
@tikurahul I think we're currently using an imgur for most of our image hosting; you should be able to access it through Passopolis

**Comment by Frijol** [comment]
This is great! I added some proofreading-type notes.
One thought – perhaps add a hook or a TL;DR to the beginning? Helpful for people to decide if they want to read all the details.

**Comment by tikurahul** [comment]
Thanks @Frijol for the comments and the great suggestions for improvement 👍

File: 2016-08-21-Crash-Reporter.md (hunk 1 above)
Comment (tikurahul) [reply 2016-08-21-Crash-Reporter.md]: Done

File: 2016-08-21-Crash-Reporter.md (hunk 2 above)
Comment (tikurahul) [reply 2016-08-21-Crash-Reporter.md]: Done.

File: 2016-08-21-Crash-Reporter.md (hunk 3 above)
Comment (tikurahul) [reply 2016-08-21-Crash-Reporter.md]: Done

File: 2016-08-21-Crash-Reporter.md (hunk 4 above)
Comment (tikurahul) [reply 2016-08-21-Crash-Reporter.md]: Done, and added notes on how to enable and disable the crash reporter.

File: 2016-08-21-Crash-Reporter.md (hunk 6 above)
Comment (tikurahul) [reply 2016-08-21-Crash-Reporter.md:27]: This should be easy enough to add. I will create a issue for myself.

File: 2016-08-21-Crash-Reporter.md (hunk 7 above)
Comment (tikurahul) [reply 2016-08-21-Crash-Reporter.md]: Done

File: 2016-08-21-Crash-Reporter.md (hunk 8 above)
Comment (tikurahul) [reply 2016-08-21-Crash-Reporter.md]: Done

**Comment by tikurahul** [comment]
Added a TL;DR section. 
Sorry about the delay in responding. I was at Tahoe 😄 .

**Comment by Frijol** [comment]
No problem. This looks great, LGTM! Feel free to publish when you're ready - and let me know if you need any help doing that

**Comment by tikurahul** [comment]
Updated image to `imgur` link.

**Comment by Frijol** [comment]
Btw - this is blog post rather that TWIT. Blog posts get posted at any time, but are not emailed out. We then call out new blog posts and link them from the emailed TWIT.

**Pull request merged**